### PR TITLE
AI Launch 2023 - Add TeachAI blog callout to /ai page

### PR DESCRIPTION
**Pull request by kelbyhawn**
Add a callout blurb and button in the hero banner on https://code.org/ai to support the launch of TeachAI.

Shown using the `teach-ai-launch-2023` DCDO flag.

**Related PRs:**
- https://github.com/code-dot-org/code-dot-org/pull/51415
- https://github.com/code-dot-org/code-dot-org/pull/51462
- https://github.com/code-dot-org/code-dot-org/pull/51564

**Jira ticket:** [ACQ-546](https://codedotorg.atlassian.net/browse/ACQ-546?atlOrigin=eyJpIjoiZjA2NWEyNTdiNjRlNDhiYjkxMGU0MDhmODBmZDE0NzciLCJwIjoiaiJ9)

----

https://user-images.githubusercontent.com/9256643/234333749-d5b7cf4c-a74e-45fc-a51f-e8850ef82ea4.mov

### CODE REVIEW

File: pegasus/sites.v3/code.org/public/css/design-system-pegasus.scss [hunk 1]
```diff
@@ -294,19 +294,21 @@ ul {
     display: flex;
     justify-content: start;
     flex-wrap: wrap;
+    gap: 1em 3em;
```
Comment (kelbyhawn) [inline pegasus/sites.v3/code.org/public/css/design-system-pegasus.scss:297]: Refactored the link list in the section under the TeachAI announcement to work w/ rtl languages (thanks @megcrenshaw!)
Screenshot 2023-04-25 at 9 06 30 AM

File: pegasus/sites.v3/code.org/public/ai.haml [hunk 2]
```diff
@@ -16,13 +16,18 @@ theme: responsive_full_width
...
 
 %section.hero-banner-basic.bg-neutral-dark{style: "display: block !important"}
   .wrapper
     .text-wrapper.col-60
       %h1{style: "color: white; margin-bottom: 0"}=hoc_s(:ai_hero_heading_01)
       %h2.heading-xxl{style: "color: white"}=hoc_s(:ai_hero_heading_02)
       %p{style: "color: white"}=hoc_s(:ai_hero_desc)
+      - if !!DCDO.get('teach-ai-launch-2023', false)
+        %a.link-button{href: "#teach-ai-callout"}
+          %i{class: "#{icon_megaphone}", style: "margin-right: 6px"}
```
Comment (kelbyhawn) [inline pegasus/sites.v3/code.org/public/ai.haml:29]: I need to make a button class that will work with icons for rtl languages bit need to think about it more and will do that in another task when there's more time ([ACQ-551](https://codedotorg.atlassian.net/browse/ACQ-551))

File: pegasus/sites.v3/code.org/public/css/design-system-pegasus.scss (hunk 1 above)
Comment (megcrenshaw) [reply pegasus/sites.v3/code.org/public/css/design-system-pegasus.scss:297]: Hurray! Looks great! This has been a huge learning curve for me too 📖 🥳

File: pegasus/sites.v3/code.org/public/ai.haml (hunk 2 above)
Comment (megcrenshaw) [reply pegasus/sites.v3/code.org/public/ai.haml:29]: Sounds good!

File: pegasus/sites.v3/code.org/public/ai.haml [hunk 3]
```diff
@@ -58,6 +63,18 @@ theme: responsive_full_width
             %p.no-margin-bottom=hoc_s(:ai_intro_list_desc_03)
     .clear
 
+%a{id: "teach-ai-callout"}
```
Comment (megcrenshaw) [inline pegasus/sites.v3/code.org/public/ai.haml]: I'm confused by this –– does this produce a whole section that's a link? Should this be inside the DCDO conditional?

File: pegasus/sites.v3/code.org/public/css/design-system-pegasus.scss [hunk 4]
```diff
@@ -294,19 +294,21 @@ ul {
...
+      gap: 0 0.75em;
 
       &::before {
         content: none;
       }
 
       i {
         font-size: 1.25em;
-        margin: 0 4px 0 0;
+        margin: 3px 0 0;
```
Comment (megcrenshaw) [inline pegasus/sites.v3/code.org/public/css/design-system-pegasus.scss:311]: Wow I didn't know what 3 arguments did until today! Cool.

**Review by megcrenshaw — APPROVED** [review]
I have one question, but not blocking! Looks great!

File: pegasus/sites.v3/code.org/public/ai.haml (hunk 3 above)
Comment (kelbyhawn) [reply pegasus/sites.v3/code.org/public/ai.haml]: It's a tag so the button in the hero banner knows where to jump to when clicked…but I just looked up [this article](https://www.w3docs.com/snippets/html/how-to-create-an-anchor-link-to-jump-to-a-specific-part-of-a-page.html) and you can add the `id` tag to elements on the page (like a `section`) that essentially does the same thing w/o needing a hanging `a` tag — I'll move all of the other `id` anchors to their respective sections and get rid of the hanging `a`s!

So much learning this week, I love it! 🤓 🙌

**Review by mgc1194 — APPROVED** [review]

**Pull request merged**